tests/posix/getdents/getdents.cpp

```diff
@@ -255,6 +255,143 @@ TEST_F(getdents, 2)
 	ASSERT_EQ(pmemfile_rmdir(pfp, "/dir1"), 0);
 }
 
+struct linux_dirent {
+	long ino;
+	off_t off;
+	unsigned short reclen;
+	char name[];
+};
+
+TEST_F(getdents, offset)
+{
+	/* create 50 files and 50 directories */
+	size_t file_dir_count = 50;
+	char path[1001];
+	PMEMfile *f = NULL;
+
+	ASSERT_TRUE(test_empty_dir(pfp, "/"));
+	memset(path, 0xff, sizeof(path));
+
+	for (size_t i = 0; i < file_dir_count; ++i) {
+		sprintf(path, "/file%04zu", i);
+		f = pmemfile_open(pfp, path, PMEMFILE_O_CREAT |
+					  PMEMFILE_O_EXCL | PMEMFILE_O_WRONLY,
+				  0644);
+		ASSERT_NE(f, nullptr) << strerror(errno);
+		pmemfile_close(pfp, f);
+
+		sprintf(path, "/dir%04zu", i);
+		ASSERT_EQ(pmemfile_mkdir(pfp, path, 0755), 0);
+	}
+
+	/* open main directory and see if lseek to end returns correct value */
+	f = pmemfile_open(pfp, "/", PMEMFILE_O_DIRECTORY | PMEMFILE_O_RDONLY);
+	ASSERT_NE(f, nullptr) << strerror(errno);
+
+	/* entry 102 - 100 files or dirs and '.' + '..' */
+	size_t max_dir_number = (file_dir_count * 2) / 14;
+	size_t max_dir_entry = (file_dir_count * 2) % 14;
+	pmemfile_off_t val = ((pmemfile_off_t)max_dir_number << 32) +
+		(pmemfile_off_t)max_dir_entry + 2;
+	ASSERT_EQ(pmemfile_lseek(pfp, f, 0, PMEMFILE_SEEK_END), val);
+
+	/*
+	 * getdents with offset 1 should return all entries except one
+	 * with 50 files and 50 directories it should return 101 entries
+	 * (50 files, 50 directories, '.' and '..')
+	 */
+	ASSERT_EQ(pmemfile_lseek(pfp, f, 1, PMEMFILE_SEEK_SET), 1);
```

Comment:
Offset value is pmemfile-specific and you are retesting it again later. Why do we even have to verify one particular offset value?